cxx4opencl/address_spaces.txt

```diff
@@ -344,3 +344,27 @@ Objects in `__constant` address space can be initialized using:
  * Literal expressions;
  * Uniform initialization syntax `{}`;
  * Using implicit constructors.
+
+==== Nested pointers
+
+{cpp} for OpenCL does not allow implicit address space conversions in nested
+pointers even with compatible address spaces. The following rules apply
+when converting between address spaces in nested pointers:
```

Comment:
Should the rationale be documented?

Reply:
I don't think we should go into this. We haven't done it for any other feature and it's better to keep it focused on just documenting the language rather than covering reasonings/etc.

I can see such things can be covered in forums or other discussions, or documents.

We have discussions in OS tooling bugs regarding this.

Reply:
Yes, maybe this is better done in a separate document if we don't think we can fit it here. I'm afraid that these reasonings will get lost quickly if there's no central document to pull all of them together.

Reply:
Right now they are all in Clang reviews, but we can certainly thinkof another document where we can explain more motivation and reasoning.